aviary/examples/run_level3_example_N3CC.py

```diff
@@ -0,0 +1,631 @@
+"""
+Sizing the N3CC using the level 3 API.
+
+Includes:
+  Takeoff, Climb, Cruise, Descent, Landing
+  Computed Aero
+  N3CC data
+"""
+
+import scipy.constants as _units
+
+import dymos as dm
+import openmdao.api as om
+from openmdao.utils.testing_utils import use_tempdirs
+from openmdao.utils.testing_utils import require_pyoptsparse
+
+from aviary.mission.flops_based.phases.energy_phase import EnergyPhase
+from aviary.mission.flops_based.phases.build_landing import Landing
+from aviary.mission.flops_based.phases.build_takeoff import Takeoff
+from aviary.subsystems.premission import CorePreMission
+from aviary.subsystems.propulsion.propulsion_builder import CorePropulsionBuilder
+from aviary.subsystems.geometry.geometry_builder import CoreGeometryBuilder
+from aviary.subsystems.mass.mass_builder import CoreMassBuilder
+from aviary.subsystems.aerodynamics.aerodynamics_builder import CoreAerodynamicsBuilder
+from aviary.subsystems.propulsion.utils import build_engine_deck
+from aviary.utils.aviary_values import AviaryValues
+from aviary.utils.functions import set_aviary_input_defaults
+from aviary.utils.functions import set_aviary_initial_values
+from aviary.utils.preprocessors import preprocess_crewpayload, preprocess_propulsion
+from aviary.utils.test_utils.default_subsystems import get_default_mission_subsystems
+from aviary.validation_cases.validation_tests import get_flops_inputs
+from aviary.variable_info.enums import LegacyCode
+from aviary.variable_info.functions import setup_trajectory_params, setup_model_options
+from aviary.variable_info.variables import Aircraft, Dynamic, Mission
+from aviary.variable_info.variable_meta_data import _MetaData as BaseMetaData
+
+
+try:
+    import pyoptsparse
+except ImportError:
+    pyoptsparse = None
+
+
+from dymos.transcriptions.transcription_base import TranscriptionBase
+
+FLOPS = LegacyCode.FLOPS
+
+# benchmark for simple sizing problem on the N3CC
+
+
+def run_trajectory(sim=True):
+    # Level 2 Equivalent: Defining an AviaryProblem() from the interface methods
+    prob = om.Problem(model=om.Group())
+
+    # Interface methods for level 2 processes drivers through the prob.add_driver
+    # function (specifying driver type, iterations, and verbosity)
+    if pyoptsparse:
+        driver = prob.driver = om.pyOptSparseDriver()
+        driver.options['optimizer'] = 'SNOPT'
+        # driver.declare_coloring()  # currently disabled pending resolve of issue 2507
```

Comment:
Issue 2507 has been fixed in OpenMDAO, so we can uncomment this line (here and anywhere else.)